core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -85,6 +85,9 @@ private[spark] class IndexShuffleBlockResolver(
   private val remoteShuffleMaxDisk: Option[Long] =
     conf.get(config.STORAGE_DECOMMISSION_SHUFFLE_MAX_DISK_SIZE)
 
+  private val checksumEnabled = conf.get(config.SHUFFLE_CHECKSUM_ENABLED)
+  private lazy val algorithm = conf.get(config.SHUFFLE_CHECKSUM_ALGORITHM)
```

Comment:
nit: Does not need to be `lazy`, right ?

Reply:
`lazy` is used because `algorithm` is not used when checksum is disabled.

Do you mean `algorithm` variable is used somewhere outside `checksumEnabled` guard? Or, simply, it's cheap operation which we don't need to put `lazy` overhead?

Reply:
The latter - it is cheap operation with no side effect to materializing it :)